client-py/iotdb/SessionPool.py

```diff
@@ -55,22 +53,23 @@ def __init__(self, pool_config: PoolConfig, max_pool_size: int, wait_timeout_in_
         self.__pool_size = 0
         self.__queue = Queue(max_pool_size)
         self.__lock = Lock()
+        self.__q_lock = Lock()
         self.__closed = False
 
     def __construct_session(self) -> Session:
-        if len(self.__pool_config.node_urls) > 0:
-            return Session.init_from_node_urls(self.__pool_config.node_urls, self.__pool_config.user_name,
-                                               self.__pool_config.password, self.__pool_config.fetch_size,
-                                               self.__pool_config.time_zone)
+        session = Session(self.__pool_config.host, self.__pool_config.port, self.__pool_config.user_name,
+                          self.__pool_config.password, self.__pool_config.fetch_size, self.__pool_config.time_zone)
+        session.open(self.__pool_config.enable_compression)
 
-        else:
-            return Session(self.__pool_config.host, self.__pool_config.ip, self.__pool_config.user_name,
-                           self.__pool_config.password, self.__pool_config.fetch_size, self.__pool_config.time_zone)
+        return session
 
     def __poll_session(self) -> Session | None:
-        if self.__queue.empty():
-            return None
-        return self.__queue.get(block=False)
+        self.__q_lock.acquire()
```

Comment:
It's better to rewrite it as
```
with self.__q_lock:
    # Do something
```
Which will automatically aquire and release lock.